### DISCUSSION TITLE
How to run specific query in different isolate?

### DISCUSSION THREAD

**Discussion by vikalpnagar** [simolus3/drift#2094] [General]
I am trying to solve a UI freeze issue when inserting ~1500 rows in one of the tables. I know we can run moor queries in different isolate. I followed [this article](https://drift.simonbinder.eu/docs/advanced-features/isolates/). I don't want every query to run on a different isolate just one SELECT query. Is it possible?

**Comment by madsane29** [comment]
Batch might solve your problem: https://drift.simonbinder.eu/docs/getting-started/writing_queries/#inserts

**Reply by vikalpnagar** [reply]
Thanks so much. Batch solved my problem. The insertion time has reduced from ~15 secs to few milliseconds.

**Comment by simolus3** [comment]
> I don't want every query to run on a different isolate just one SELECT query. Is it possible?

It is not directly possible. A drift isolate is not just for running queries, it also manages stream query updates between all isolates. So you can't easily run some specific operations in one isolate and others in another.

You could use a [MultiExecutor](https://pub.dev/documentation/drift/latest/drift/MultiExecutor-class.html) to generally run inserts and updates in a background isolate while running selects in the main isolate. Perhaps that's what you're looking for?

If you only need to run one single operation in a background isolate, I recommend not using isolates at all. Instead, use Flutter's `compute` to create a new and totally independent drift database instance in the background isolate that is not talking to the main database instance at all. Then, do the expensive insert there and finally call [notifyUpdates](https://pub.dev/documentation/drift/latest/drift/DatabaseConnectionUser/notifyUpdates.html) so that streams that might be invalidated by that write change.

**Reply by vikalpnagar** [reply]
Thanks for the reply @simolus3. Batch solved my problem as suggested by @madsane29.
